src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -67,25 +84,46 @@ public Path which(String fn) {
 
     @Override
     public String[] getCommand() {
-        String[] decorated = cmds;
+        String[] decorated = Arrays.copyOf(cmds, cmds.length);
         if (shellDecorator != null) {
             decorated = shellDecorator.decorate(decorated);
         }
-        // First item in the command is the executable. If it's given as absolute path, add quotes around it
-        // in case the path contains space which will break the whole command line
-        // See security remarks:
-        // https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-createprocesswithlogonw
-        if (isAbsolutePath(decorated[0])) {
-            decorated[0] = String.format("\"%s\"", decorated[0]);
-        }
         return decorated;
     }
 
     @Override
     protected Process createProcess() throws IOException {
+        if (needToSwitchUser()) {
+            return createRunasProcess();
+        } else {
+            ProcessBuilder pb = new ProcessBuilder();
+            pb.environment().putAll(environment);
+            return pb.directory(dir).command(getCommand()).start();
+        }
+    }
+
+    private Process createRunasProcess() throws IOException {
+        String username = userDecorator.getUser();
+
+        byte[] credBlob = WindowsCredUtils.read(username);
+        ByteBuffer bb = ByteBuffer.wrap(credBlob);
+        CharBuffer cb = StandardCharsets.UTF_8.decode(bb);
+
+        List<String> args = new ArrayList<>();
+        args.add(runasExePath);
+        args.add("-u:" + username);
+        args.add("-p:" + cb);
+        args.add("-l:off");  // disable logging
+        args.addAll(Arrays.asList(getCommand()));
+
+        Arrays.fill(cb.array(), (char) 0);  // zero-out temporary buffers
+        Arrays.fill(bb.array(), (byte) 0);
+
         ProcessBuilder pb = new ProcessBuilder();
         pb.environment().putAll(environment);
-        return pb.directory(dir).command(getCommand()).start();
+        Process p = pb.directory(dir).command(args).start();
+        args.clear();  // best effort to clear password presence
+        return p;
```

Comment:
follow the logging behavior in UnixExec. Log the commands that we're executing (but ensure you do not include the call to runas as that may log the password)